### PR TITLE
fix: erc721 erc165 support

### PR DESCRIPTION
**Pull request by fubuloubu**
### What I did
Currently, the example in the Vyper org for ERC721 is not 100% compliant because it doesn't support the ERC165 method_id `supportsInterface(bytes4)` (because we don't currently have `bytes4` type)

### How I did it
Hack foo in the fallback function

### How to verify it
Tests pass, also deploying to ethereum and check OpenSea for compliance

### Description for the changelog
Fix ERC721 example for compliance

### Cute Animal Picture
fun penguin

### CODE REVIEW

File: examples/tokens/ERC721.vy [hunk 1]
```diff
@@ -360,3 +356,15 @@ def burn(_tokenId: uint256):
     self._clearApproval(owner, _tokenId)
     self._removeTokenFrom(owner, _tokenId)
     log Transfer(owner, ZERO_ADDRESS, _tokenId)
+
+
+@external
+def __default__() -> bool:
```
Comment (skellet0r) [inline examples/tokens/ERC721.vy:362]: Wooo

**Comment by skellet0r** [comment]
So much better than using `pizza_mandate_apology(uint256)`

**Comment by fubuloubu** [comment]
> So much better than using `pizza_mandate_apology(uint256)`

lololol did that work????

File: examples/tokens/ERC721.vy [hunk 2]
```diff
@@ -66,34 +66,33 @@ ownerToOperators: HashMap[address, HashMap[address, bool]]
...
 
-@view
+@pure
 @external
-def supportsInterface(_interfaceID: bytes32) -> bool:
+#def supportsInterface(interface_id: bytes4) -> bool:
+def pizza_mandate_apology(interface_id_int: uint256) -> bool:
     """
     @dev Interface identification is specified in ERC-165.
     @param _interfaceID Id of the interface
```
Comment (fubuloubu) [inline examples/tokens/ERC721.vy:91]:
```suggestion
    @param interface_id Id of the interface
```

**Comment by SuperCipher** [comment]
@fubuloubu 
I have tried to [deploy](https://rinkeby.etherscan.io/address/0xb3af2fbea307f2f222b013c973c56271dcc9111d) the `pizza_mandate_apology(uint256)` ERC721 hack on Opensea Rinkeby testnet but neither Opensea nor the Ethscan recognizes it as ERC721 instead Ethscan [recognized it as ERC-20](https://rinkeby.etherscan.io/token/0xb3af2fbea307f2f222b013c973c56271dcc9111d).
 After I change it to [`__default__`  hack](https://discord.com/channels/809793915578089484/809794150550732830/935564562600689695) and deploy on both Opensea Rinkeby testnet and Polygon Mumbai testnet. Both of them recognized it as ERC721 now.
### Here is the working deployed example with the Vyper ERC721 example using `__default__`  hack
On Rinkeby.
https://testnets.opensea.io/assets/0x9f63a231dfb532f4459c4a9579fb6dd25eb6f180/2
https://rinkeby.etherscan.io/address/0x9f63a231dfb532f4459c4a9579fb6dd25eb6f180

Here is deployed on Polygon testnet Mumbai.
https://testnets.opensea.io/assets/mumbai/0x1fbbba6f55168b825d26fcfef9738ac53dc03cc1/1
https://mumbai.polygonscan.com/address/0x1fbbba6f55168b825d26fcfef9738ac53dc03cc1

**Comment by fubuloubu** [comment]
> @fubuloubu I have tried to [deploy](https://rinkeby.etherscan.io/address/0xb3af2fbea307f2f222b013c973c56271dcc9111d) the `pizza_mandate_apology(uint256)` ERC721 hack on Opensea Rinkeby testnet but neither Opensea nor the Ethscan recognizes it as ERC721 instead Ethscan [recognized it as ERC-20](https://rinkeby.etherscan.io/token/0xb3af2fbea307f2f222b013c973c56271dcc9111d). After I change it to [`__default__` hack](https://discord.com/channels/809793915578089484/809794150550732830/935564562600689695) and deploy on both Opensea Rinkeby testnet and Polygon Mumbai testnet. Both of them recognized it as ERC721 now.
> 
> ### Here is the working deployed example with the Vyper ERC721 example using `__default__` hack
> On Rinkeby. https://testnets.opensea.io/assets/0x9f63a231dfb532f4459c4a9579fb6dd25eb6f180/2 https://rinkeby.etherscan.io/address/0x9f63a231dfb532f4459c4a9579fb6dd25eb6f180
> 
> Here is deployed on Polygon testnet Mumbai. https://testnets.opensea.io/assets/mumbai/0x1fbbba6f55168b825d26fcfef9738ac53dc03cc1/1 https://mumbai.polygonscan.com/address/0x1fbbba6f55168b825d26fcfef9738ac53dc03cc1

That is strange, we had someone who was able to get it to recognize on OpenSea w/ the `pizza_mandate_apology` hack on Rinkeby

**Comment by fubuloubu** [comment]
> @fubuloubu 
> I have tried to [deploy](https://rinkeby.etherscan.io/address/0xb3af2fbea307f2f222b013c973c56271dcc9111d) the `pizza_mandate_apology(uint256)` ERC721 hack on Opensea Rinkeby testnet but neither Opensea nor the Ethscan recognizes it as ERC721 instead Ethscan [recognized it as ERC-20](https://rinkeby.etherscan.io/token/0xb3af2fbea307f2f222b013c973c56271dcc9111d).
>  After I change it to [`__default__`  hack](https://discord.com/channels/809793915578089484/809794150550732830/935564562600689695) and deploy on both Opensea Rinkeby testnet and Polygon Mumbai testnet. Both of them recognized it as ERC721 now.
> ### Here is the working deployed example with the Vyper ERC721 example using `__default__`  hack
> On Rinkeby.
> https://testnets.opensea.io/assets/0x9f63a231dfb532f4459c4a9579fb6dd25eb6f180/2
> https://rinkeby.etherscan.io/address/0x9f63a231dfb532f4459c4a9579fb6dd25eb6f180
> 
> Here is deployed on Polygon testnet Mumbai.
> https://testnets.opensea.io/assets/mumbai/0x1fbbba6f55168b825d26fcfef9738ac53dc03cc1/1
> https://mumbai.polygonscan.com/address/0x1fbbba6f55168b825d26fcfef9738ac53dc03cc1

Oh I know what your issue is here. Before verifying, edit your ABI file to change `pizza_mandate_apology` to `supportsInterface` and  `interface_id` to be `bytes4`.

The code is fine, it's literally is the same as doing it in the fallback, but the ABI is probably red flagging you on Etherscan and OpenSea

**Review by charles-cooper — APPROVED** [review]
approved, will open a separate PR to upgrade to bytes4

**Pull request merged**